base/linalg/bunchkaufman.jl

```diff
@@ -41,16 +41,13 @@ function bkfact!(A::StridedMatrix{<:BlasFloat}, rook::Bool = false)
 end
 
 """
-    bkfact(A, uplo::Symbol=:U, symmetric::Bool=issymmetric(A), rook::Bool=false) -> BunchKaufman
+    bkfact(A, rook::Bool=false) -> BunchKaufman
 
-Compute the Bunch-Kaufman [^Bunch1977] factorization of a symmetric or Hermitian
-matrix `A` and return a `BunchKaufman` object.
-`uplo` indicates which triangle of matrix `A` to reference.
-If `symmetric` is `true`, `A` is assumed to be symmetric. If `symmetric` is `false`,
-`A` is assumed to be Hermitian. If `rook` is `true`, rook pivoting is used. If
-`rook` is false, rook pivoting is not used.
-The following functions are available for
-`BunchKaufman` objects: [`size`](@ref), `\\`, [`inv`](@ref), [`issymmetric`](@ref), [`ishermitian`](@ref).
+Compute the Bunch-Kaufman [^Bunch1977] factorization of a symmetric or Hermitian matrix `A` as ``PUDU'P'`` or ``PLDL'P'``, depending on which triangle is stored in `A`, and return a `BunchKaufman` object.
```

Comment:
The permutations are symmetric so you can leave out the apostrophe after the `P`s.

Reply:
This bothered me too, for another reason: to be consistent with the use of permutation matrices, I think I should consider writing
```julia
LDL' = PAP'
```
but in this specific case `P'=P⁻¹=P`, so basically anything goes.

[R, for example](http://stat.ethz.ch/R-manual/R-patched/library/Matrix/html/BunchKaufman-methods.html), uses `A = P LDL' P'`, so that's one way to go.

I thought about mentioning that `P` is symmetric and its own inverse in the docstring.

What do you think? I like the symmetry of the formula, but I am happy to make any changes you suggest.

Reply:
It is redundant and leaving out the apostrophe clearly shows the structure of the factorization. I'm fine with mentioning it explicitly in the text as well.

Reply:
Done, thanks for the comments.